utils.go

```diff
@@ -108,6 +111,67 @@ func ExistOnlyOne(arr []string) bool {
 	}
 	return exist
 }
+func IterateAddr(addr string) <-chan Host {
+	hostChan := make(chan Host)
+	_, _, err := net.ParseCIDR(addr)
+	if err == nil {
+		// is CIDR
+		return Iterate(strings.NewReader(addr))
+	}
+	ip := net.ParseIP(addr)
+	if ip == nil {
+		ip, err = LookupIP(addr)
+		if err != nil {
+			close(hostChan)
+			slog.Error("Not a valid IP, IP CIDR or domain", "addr", addr)
+			return hostChan
+		}
+	}
+	go func() {
+		slog.Info("Enable infinite mode", "init", ip.String())
+		lowIP := ip
+		highIP := ip
+		hostChan <- Host{
+			IP:     ip,
+			Origin: addr,
+			Type:   HostTypeIP,
+		}
+		for i := 0; i < math.MaxInt; i++ {
+			if i%2 == 0 {
```

Comment:
Thanks for your code. I was thinking about using channel this way. But I was worry about grow memory too fast. Isn't all these ips not yet processed in channel taking memory?

Reply:
So it's an unbuffered channel for hostChan. It will block at the sending action until a scanning goroutine picks one away from it. Not consuming memory more than 1 Host instance.

Reply:
I understand now. It is indeed better. Thanks again!